bpf/bpf_host.c

```diff
@@ -448,8 +447,7 @@ resolve_srcid_ipv4(struct __ctx_buff *ctx, __u32 srcid_from_proxy,
 	/* If we could not derive the secctx from the packet itself but
 	 * from the ipcache instead, then use the ipcache identity.
 	 */
-	else if (identity_from_ipcache_ok() &&
-		 !identity_is_reserved(srcid_from_ipcache))
+	else if (identity_from_ipcache_ok())
 		src_id = srcid_from_ipcache;
```

Comment:
@cilium/sig-datapath the IPv4 vs. IPv6 cases here are divergent, seems like one checks for `WORLD_ID` and the other doesnt?

Reply:
Yes, we have a special case for IPv6. See `derive_src_id`. The difference is:
- IPv4: No need to check the value of `src_id`. It was initialized with `WORLD_ID` and not modified before hitting this condition.
- IPv6: The `src_id` value may have been changed by `derive_src_id`.